src/game/file_io.c

```diff
@@ -855,14 +827,16 @@ static int load_scenario_to_buffers(const char *filename, int *version)
     }
     *version = get_scenario_version(fp);
     init_scenario_data(*version);
+    memblock compress_buffer;
+    core_memblock_init(&compress_buffer, COMPRESS_BUFFER_INITIAL_SIZE);
```

Comment:
An inconsistency: for scenarios you init and free the memory block in this function, while in savegame reading you do it on the upper-level function.

Why not do it at the same logical place for both cases?

Reply:
The memory blocks were created in the same place the files were read, so I'd argue that's the real inconsistency here. However I've gone ahead and changed it.

Reply:
Makes sense. I only noticed it for the memory block because the change is highlighted.

Reply:
Any thoughts on whether the file should be opened in the higher function vs the lower function? 🤔

Reply:
I think it makes more sense in the lower function as the variable isn't needed on the higher function.